fix (@status-go/stickers) Stickers still show in the recent list after being uninstalled

kounkou opened the pull request:
### Description

Stickers still show in the recent list after being uninstalled

fixes #12677
linked to https://github.com/status-im/status-desktop/pull/13066

Important changes:
- Updates the  private function `recentStickers` private function to filter the stickers based on if the sticker pack is installed or not. Before, the filter was not applied and returned a list that didn't take into account if the pack was installed, therefore causing a bug.

kounkou committed:
fix (@status-go/stickers) Stickers still show in the recent list after being uninstalled

fixes #12677

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (22)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 00b7a4c8 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4474/1/) | 2023-12-15 16:46:25 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231215-164312-00b7a4-pr4474.zip) |
| :heavy_check_mark: | 00b7a4c8 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4474/1/) | 2023-12-15 16:48:58 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231215-164308-00b7a4-pr4474.aar) |
| :heavy_check_mark: | 00b7a4c8 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/1/) | 2023-12-15 17:18:47 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/1//consoleText) |
| :x: | 00b7a4c8 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4474/3/) | 2023-12-15 17:20:55 | ~14 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4474/3/consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 984c741f | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/2/) | 2023-12-20 10:13:00 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/2//consoleText) |
| :heavy_check_mark: | 984c741f | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4474/2/) | 2023-12-20 10:13:13 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231220-101033-984c74-pr4474.aar) |
| :heavy_check_mark: | 984c741f | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4474/2/) | 2023-12-20 10:14:03 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231220-101033-984c74-pr4474.zip) |
| | | | | | | |
| :heavy_check_mark: | 826889bd | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4474/3/) | 2023-12-20 10:19:36 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231220-101751-826889-pr4474.zip) |
| :heavy_multiplication_x: | 826889bd | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/3/) | 2023-12-20 10:20:15 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/3//consoleText) |
| :heavy_check_mark: | 826889bd | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4474/3/) | 2023-12-20 10:20:20 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231220-101748-826889-pr4474.aar) |
| | | | | | | |
| :heavy_check_mark: | 9ce86ae8 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4474/4/) | 2023-12-20 10:32:06 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231220-103003-9ce86a-pr4474.aar) |
| :heavy_multiplication_x: | 9ce86ae8 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/4/) | 2023-12-20 10:33:21 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/4//consoleText) |
| :heavy_check_mark: | 9ce86ae8 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4474/4/) | 2023-12-20 10:33:52 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231220-103003-9ce86a-pr4474.zip) |
| :heavy_check_mark: | 9ce86ae8 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4474/7/) | 2023-12-20 10:34:51 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231220-103115-9ce86a-pr4474.zip) |
| | | | | | | |
| :heavy_multiplication_x: | f601128b | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/5/) | 2023-12-20 10:38:15 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/5//consoleText) |
| :heavy_check_mark: | f601128b | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4474/5/) | 2023-12-20 10:39:06 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231220-103543-f60112-pr4474.zip) |
| :heavy_check_mark: | f601128b | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4474/5/) | 2023-12-20 10:41:28 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231220-103543-f60112-pr4474.aar) |
| :heavy_multiplication_x: | f601128b | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/6/) | 2023-12-20 10:54:08 | ~40 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/6//consoleText) |
| :heavy_multiplication_x: | f601128b | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/7/) | 2023-12-20 18:57:13 | ~46 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/7//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 319bdaf7 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/8/) | 2023-12-20 19:56:19 | ~53 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/8//consoleText) |
| :heavy_check_mark: | 319bdaf7 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4474/6/) | 2023-12-20 19:56:38 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231220-195525-319bda-pr4474.zip) |
| :heavy_check_mark: | 319bdaf7 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4474/6/) | 2023-12-20 19:56:59 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231220-195521-319bda-pr4474.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | d5f7760c | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/9/) | 2023-12-20 20:02:04 | ~48 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/9//consoleText) |
| :heavy_check_mark: | d5f7760c | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4474/7/) | 2023-12-20 20:02:55 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231220-200111-d5f776-pr4474.zip) |
| :heavy_check_mark: | d5f7760c | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4474/7/) | 2023-12-20 20:03:03 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231220-200111-d5f776-pr4474.aar) |
| | | | | | | |
| :heavy_check_mark: | 6858bf44 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4474/8/) | 2023-12-20 20:16:06 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231220-201441-6858bf-pr4474.zip) |
| :heavy_check_mark: | 6858bf44 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4474/8/) | 2023-12-20 20:16:12 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231220-201441-6858bf-pr4474.aar) |
| :x: | 6858bf44 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4474/12/) | 2023-12-20 20:37:51 | ~13 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4474/12/consoleText) |
| :heavy_check_mark: | 6858bf44 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/10/) | 2023-12-20 20:48:16 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4474/10//consoleText) |
| :heavy_check_mark: | 6858bf44 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4474/16/) | 2023-12-21 01:27:22 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231221-012402-6858bf-pr4474.zip) |

kounkou commented:
I updated a private function. Checking if adding a unittest is possible as there's no current infrastructure for unit testing for the entire file.

igor-sirotin commented:
> Checking if adding a unittest is possible

A test would be good!

kounkou commented:
> > Checking if adding a unittest is possible
> 
> A test would be good!

Yes, I am not going to merge until I have a decent one. I wanted to put the PR out for now as a respect to people precious time. I am adding it by ~~EOD~~ Monday 👍🏾  (on my time LOL)

kounkou commented:
I believed this PR required special attention for testing, so I updated the code to be able to mock the functions `GetInstalledStickerPacks` and `GetRecentStickers` insisde the `func (api *API) recentStickers() ([]Sticker, error) {` function. 
While this took me a day or 2 to make sure there's no regression, I think the entire codebase should be refactored similarly to be able to easily enable capacity to mock the function, hence to create more testable code in the future.
On a another perspective, we should strive for TDD to help produce testable code.

-----

- Fixed codeClimate issues due to high number of function mocks
- Split the mocks into contextual mocks files

kounkou commented on multiaccounts/settings/database_settings_manager.go:
```diff
@@ -0,0 +1,77 @@
+package settings
```
This interface should have been split to make the code more easily testable.
Code smell can be seen when seeing in go struct some 

```
*settings.Database
```
As this means we are forcing the developer/tester to pass an actual implementation of the Database. This is really sub-optimal because it can take some time to setup all the parameters required to run the application.
Instead, this PR proposal is to use an interface and be able to mock the critical function that are being called. 
With this PR we do not go level to call the database, but we mock the call hence we control the result from the low level API call.
As a result, we can avoid flaky tests, and have fast tests. 

To finish, it might be necessary to apply similar pattern for other portions of the code. But this is not in the scope of this PR

kounkou commented on multiaccounts/accounts/database.go:
```diff
@@ -283,7 +283,7 @@ func (a *Keypair) Operability() AccountOperable {
 // Database sql wrapper for operations with browser objects.
 type Database struct {
-	*settings.Database
+	settings.DatabaseSettingsManager
```
In a separate task, we can make the Database take different dependencies like Reader, Writer,... for the Database to be designed better.
I leave this out of the scope of this ticket.

borismelnik reviewed (approved):
LGTM :)
